README.md

```diff
@@ -1,58 +1,125 @@
-- 🚀 Fully fledged Python project setup with linting and formatting.
+## 🌟 Key Features
 
-## 📦 Requirements
+- 🔒 Safely execute an arbitrary Python Code via [BeeAI Code Interpreter](https://github.com/i-am-bee/beeai-code-interpreter).
+- ⚡ Fully fledged Python project setup with linting and formatting.
 
-- Python version 3.11 or higher
-- Poetry, a tool for Python package management. Installation steps can be found [here](https://python-poetry.org/docs/#installation) 
-- Container system like [Rancher Desktop](https://rancherdesktop.io/), [Podman](https://podman.io/) (VM must be rootfull machine) or [Docker](https://www.docker.com/).
-- LLM Provider either external [WatsonX](https://www.ibm.com/watsonx) (OpenAI, Groq, ...) or local [ollama](https://ollama.com).
+---
 
-## 🛠️ Getting started
+## 📋 Requirements
 
-1. Clone this repository or [use it as a template](https://github.com/new?template_name=beeai-framework-py-starter&template_owner=i-am-bee).
-2. Install dependencies `poetry install`.
-3. Install and start the poetry environment `poetry self add poetry-plugin-shell` and `poetry shell`
-4. Configure your project:
-    1. Make a copy of `.env.example` and rename the copy to `.env`
-    2. Fill in missing values in the `.env` file (default LLM provider is locally hosted `Ollama`).
-5. Run the agent `python beeai_framework_starter/agent.py`
+- **Python 3.11+**
+- **Poetry** for Python package management - See [installation guide](https://python-poetry.org/docs/#installation) 
+- **Container system** (with Compose support):
+    - [Docker](https://www.docker.com/)
+    - [Rancher](https://www.rancher.com/) (For macOS, use VZ instead of QEMU)
+    - [Podman](https://podman.io/) (Requires [Compose](https://podman-desktop.io/docs/compose/setting-up-compose) and rootful machine)
+- **LLM Provider** - External [WatsonX](https://www.ibm.com/watsonx) (OpenAI, Groq, ...) or local [Ollama](https://ollama.com)
+- **IDE/Code Editor** (e.g., WebStorm, VSCode) - Optional but recommended for smooth smooth configuration handling
 
-To run an agent with a custom prompt, simply do this `python beeai_framework_starter/agent.py <<< 'Hello Bee!'`
+---
 
-🧪 More examples can be found [here](https://github.com/i-am-bee/beeai-framework/blob/main/python/examples).
+## 🛠️ Installation
 
-> [!TIP]
->
-> To use Bee agent with [Python Code Interpreter](https://github.com/i-am-bee/bee-code-interpreter) refer to the [Code Interpreter](#code-interpreter) section.
+**Step one:** Clone this repository or [use it as a template](https://github.com/new?template_name=beeai-framework-py-starter&template_owner=i-am-bee)
+```sh
+git clone https://github.com/i-am-bee/beeai-framework-py-starter.git
+cd beeai-framework-py-starter
+```
 
-## 🏗 Infrastructure
+**Step two:** Create and activate a virtual environment
+```sh
+python -m venv .venv  
+source .venv/bin/activate
+```
```

Comment:
This step shouldn't be necessary, IIUC `poetry shell` will actually a new vent for you if one isn't found